spec/models/crop_spec.rb

```diff
@@ -419,164 +419,135 @@
         expect(tomato.scientific_names.size).to eq 2
-        tomato.add_scientific_names_from_csv("Baz,   Quux") # multiple spaces
-        expect(tomato.scientific_names.size).to eq 4
+        expect(tomato.scientific_names[0].name).to eq "Baz"
+        expect(tomato.scientific_names[1].name).to eq "Quux"
       end
     end # scientific names
 
     context "alternate names" do
       it "loads an alternate name" do
-        tomato = FactoryGirl.create(:tomato)
-        expect(tomato.alternate_names.size).to eq 0
-        tomato.add_alternate_names_from_csv("Foo")
+        row = ["tomato", "http://en.wikipedia.org/wiki/Parent", "", "", "Foo"]
+        tomato = CsvImporter.new.import_crop(row)
         expect(tomato.alternate_names.size).to eq 1
         expect(tomato.alternate_names.last.name).to eq "Foo"
       end
 
-      it "doesn't load duplicate alternate names" do
-        tomato = FactoryGirl.create(:tomato)
-        expect(tomato.alternate_names.size).to eq 0
-        tomato.add_alternate_names_from_csv("Foo")
-        expect(tomato.alternate_names.size).to eq 1
-        tomato.add_alternate_names_from_csv("Foo")
-        expect(tomato.alternate_names.size).to eq 1 # still 1, doesn't add another
-      end
-
       it "adds multiple alternate names" do
-        tomato = FactoryGirl.create(:tomato)
-        expect(tomato.alternate_names.size).to eq 0
-        tomato.add_alternate_names_from_csv("Foo, Bar")
+        row = ["tomato", "http://en.wikipedia.org/wiki/Parent", "", "", "Foo, Bar"]
+        tomato = CsvImporter.new.import_crop(row)
         expect(tomato.alternate_names.size).to eq 2
         expect(tomato.alternate_names[0].name).to eq "Foo"
         expect(tomato.alternate_names[1].name).to eq "Bar"
       end
 
       it "adds multiple alt names with variant spacing" do
-        tomato = FactoryGirl.create(:tomato)
-        expect(tomato.alternate_names.size).to eq 0
-        tomato.add_alternate_names_from_csv("Foo,Bar") # no space
-        expect(tomato.alternate_names.size).to eq 2
-        tomato.add_alternate_names_from_csv("Baz,   Quux") # mutliple spaces
+        row = ["tomato", "http://en.wikipedia.org/wiki/Parent", "", "", "Foo,Bar,Baz,   Quux"]
+        tomato = CsvImporter.new.import_crop(row)
         expect(tomato.alternate_names.size).to eq 4
+        expect(tomato.alternate_names[0].name).to eq "Foo"
+        expect(tomato.alternate_names[1].name).to eq "Bar"
+        expect(tomato.alternate_names[2].name).to eq "Baz"
+        expect(tomato.alternate_names[3].name).to eq "Quux"
+      end
+
+      it "doesn't add a duplicate alternate name" do
+        row = ["tomato", "http://en.wikipedia.org/wiki/tomato", "", "", "Foo"]
+        tomato = CsvImporter.new.import_crop(row)
+        row = ["tomoto", "http://en.wikipedia.org/wiki/tomoto", "", "", "Foo"]
+        tomoto = CsvImporter.new.import_crop(row)
+        expect(tomato.alternate_names.size).to eq 1
+        expect(tomoto.alternate_names.size).to eq 0
```

Comment:
Wait, no, we actually assume in the code that every AlternateName belongs to exactly one Crop. So what's happening here is that the second crop grabs the AlternateName from its former owner. We probably don't want to allow that.